src/www/ui/api/documentation/openapi.yaml

```
@@ -612,6 +612,96 @@ paths:
         default:
           $ref: '#/components/responses/defaultResponse'
 
+  /uploads/{id}/permissions:
+    parameters:
+      - name: id
+        required: true
+        description: Enter any upload id of that folder
+        in: path
+        schema:
+          type: integer
+    put:
+      operationId: setUploadPermissions
+      tags:
+        - Upload
+      summary: Set permissions for a upload in a folder for different groups
+      description: >
+        Set permissions for a upload in a folder for different groups
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                folderId:
+                  type: integer
+                  description: Enter a folder id
+                allUploadsPermission:
+                  type: string
+                  enum:
+                  - yes
+                  - no
+                  description: Apply same permissions to all uploads in this folder
+                newGroupId:
+                  type: integer
+                  description: Enter the id of the group you want to add or edit permission for this upload
+                newPermission:
+                  type: string
+                  enum:
+                  - none
+                  - read
+                  - write
```

Please change values to be consistent with other endpoints.
```suggestion
                  - read_only
                  - read_write
```

---

Yes, got it